File: src/features/tickets/models/ticket.ts
import type { Category, TicketStatus } from '@/core/config/constants'

export type { Category, TicketStatus }

export interface Ticket {
  id: string
  subject: string
  description: string
  predicted_category: Category
  confidences: Record<Category, number>
  final_category: Category | null
  status: TicketStatus
  created_by: string
  assignee_id: string | null
  created_at: string
  updated_at: string
}

export interface CreateTicketDto {
  subject: string
  description: string
}

export interface UpdateTicketDto {
  status?: TicketStatus
  final_category?: Category
  assignee_id?: string
}

File: backend/app/models/ticket.py
from pydantic import BaseModel


class CreateTicketRequest(BaseModel):
    subject: str
    description: str


class UpdateTicketRequest(BaseModel):
    status: str | None = None
    final_category: str | None = None
    assignee_id: str | None = None


class TicketResponse(BaseModel):
    id: str
    subject: str
    description: str
    predicted_category: str
    confidences: dict[str, float]
    final_category: str | None
    status: str
    created_by: str
    assignee_id: str | None
    created_at: str
    updated_at: str


class PaginatedTickets(BaseModel):
    items: list[TicketResponse]
    total: int
    page: int
    size: int
    pages: int
